lib/matplotlib/colorbar.py

```diff
@@ -357,34 +426,119 @@ def draw_all(self):
     def config_axis(self):
         ax = self.ax
         if self.orientation == 'vertical':
-            ax.xaxis.set_ticks([])
             # location is either one of 'bottom' or 'top'
             ax.yaxis.set_label_position(self.ticklocation)
             ax.yaxis.set_ticks_position(self.ticklocation)
+            if (isinstance(self.norm, colors.LogNorm)
+                    and self._use_adjustable()):
+                ax.set_xscale('log')
+                ax.set_yscale('log')
+            ax.xaxis.set_ticks([])
+            ax.xaxis.set_ticks([], minor=True)
+
         else:
-            ax.yaxis.set_ticks([])
             # location is either one of 'left' or 'right'
             ax.xaxis.set_label_position(self.ticklocation)
             ax.xaxis.set_ticks_position(self.ticklocation)
+            if (isinstance(self.norm, colors.LogNorm)
+                    and self._use_adjustable()):
+                ax.set_xscale('log')
+                ax.set_yscale('log')
+            ax.yaxis.set_ticks([])
+            ax.yaxis.set_ticks([], minor=True)
```

Comment:
The bulk of this method can be shortened and clarified:
```python
if isinstance(self.norm, colors.LogNorm)
                     and self._use_adjustable()):
    ax.set_xscale('log')
    ax.set_yscale('log')
if self.orientation == 'vertical':
    long_axis, short_axis = ax.yaxis, ax.xaxis
else:
    long_axis, short_axis = ax.xaxis, ax.yaxis
long_axis.set_label_position(self.ticklocation)
long_axis.set_ticks_position(self.ticklocation)
short_axis.set_ticks([])
short_axis.set_ticks([], minor=True)
```

Reply:
Great idea.  Thanks!